built_value_generator/lib/src/value_source_class.dart

```diff
@@ -29,6 +31,11 @@ abstract class ValueSourceClass
       new _$ValueSourceClass._(element: element);
   ValueSourceClass._();
 
+  @memoized
+  ParsedLibraryResult get parsedLibrary =>
+      // ignore: deprecated_member_use
+      ParsedLibraryResultImpl.tmp(element.library);
```

Comment:
This is different from what was proposed in my internal CL.
Semantically it is the same, but performance is not the same.

Here you create a new `ParsedLibraryResult` instance every time, which includes parsing all library files again. This is a non-trivial amount of work. If you want it to be more efficient, create `ParsedLibraryResult` and pass it around.

Reply:
Thanks. Note that `@memoized` means it's _somewhat_ cached; it's only fetched once per containing class. This does mean we fetch more often. But it's neater ;)

I checked performance on the biggest built_value-containing package we have--82 classes--and there was no measurable slowdown. If anything this is a little faster than what we had before.

Once we're off the final solution I'll see if there's any further speedup to be had.

Thanks!